docs/devguide.rst

```diff
@@ -1,4 +1,288 @@
 Developer Guide
 ===============
 
-**COMING SOON**
+
+
+Releasing new versions
+----------------------
+
+Releasing new versions can be done in two different ways
+
+* automatically
+* manually
+
+In this document I'll outline the steps required for manual release.
+
+Version commit
+~~~~~~~~~~~~~~
+
+The version commit is the first step in releasing a new version. This can be done
+in a later step, but for ease-of-release I'm putting it at the beginning.
+
+To do this, first decide that you want to release a new version and that **all**
+commits for that release are in the branch you're interested in releasing. We do
+all work on the `development` branch.
+
+Good, you're now ready.
+
+The final commit will happen now and it is a simple one. You need to change the
+following file.
+
+* `f5/__init__.py`
+
+In this file is a single line that specifies the current version of the SDK. For
+example,
+
+.. code-block:: python
+
+   # Copyright 2016 F5 Networks Inc.
+   #
+   # Licensed under the Apache License, Version 2.0 (the "License");
+   # you may not use this file except in compliance with the License.
+   # You may obtain a copy of the License at
+   #
+   #    http://www.apache.org/licenses/LICENSE-2.0
+   #
+   # Unless required by applicable law or agreed to in writing, software
+   # distributed under the License is distributed on an "AS IS" BASIS,
+   # WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+   # See the License for the specific language governing permissions and
+   # limitations under the License.
+   #
+   __version__ = '3.0.1'
+
+Your job is to increment that number wisely. What is "wisely"? It means that it is
+your duty to know if the fixes and additions in the release warrant a,
+
+* major version change
+* minor version change
+* patch version change
+
+The details of each of these is described here
+
+* http://semver.org/
+
+Let's assume that we only had a minor release. In that situation, we would change
+the version line to look like this.
+
+.. code-block:: python
+
+   __version__ = '3.0.2'
+
+With this change made, add it and commit it back to Github. This is the last commit
+before releasing.
+
+Additionally, since this is the last commit, it is important that you *tag* this
+release at the current commit. Therefore, you should additionally issue this
+command,
+
+.. code-block:: bash
+
+   git tag v3.0.2
+
+And also push that tag upstream because it is what you will tell Github to release
+from.
+
+.. code-block:: bash
+
+   git push v3.0.2 upstream
```

Comment:
Strictly speaking this is unnecessary since the release notes interface adds a tag.